src/Plugin.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Hostnet\Component\Path;
+
+use Composer\Composer;
+use Composer\IO\IOInterface;
+use Composer\Plugin\PluginInterface;
+use Composer\Script\ScriptEvents;
+
+class Plugin implements PluginInterface
+{
+    /**
+     * @var Composer
+     */
+    private $composer;
+
+    /**
+     * @var IOInterface
+     */
+    private $io;
+
+    public static function getSubscribedEvents()
+    {
+        return [ScriptEvents::PRE_AUTOLOAD_DUMP => 'onPreAutoloadDump'];
+    }
+
+    public function activate(Composer $composer, IOInterface $io)
+    {
+        $this->composer = $composer;
+        $this->io       = $io;
+    }
+
+    public function onPreDumpAutoload()
+    {
+        $vendor_dir = realpath($this->composer->getConfig()->get('vendor-dir'));
+        $base_dir   = getcwd();
```

Comment:
composer changes working directory in [Application](https://github.com/composer/composer/blob/57e33fcad7a4ce57363bf3ae72425d3cebd61cb3/src/Composer/Console/Application.php#L136
)

It will only run with a `composer.json` in the current directory and for the vendor folder (and with the plugins) defined in that `composer.json` or the directory passed to the `--working-dir|-d` option.